integration/nwo/common/docker/utils.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package docker
+
+import (
+	"net"
+	"strings"
+	"sync"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/flogging"
+	"github.com/pkg/errors"
+)
+
+var logger = flogging.MustGetLogger("fsc.integration.fabric")
+
+type Docker struct {
+	Client *docker.Client
+}
+
+var singleInstance *Docker
+var lock = &sync.Mutex{}
+
+func GetInstance() (*Docker, error) {
+
+	if singleInstance == nil {
+		lock.Lock()
+		defer lock.Unlock()
+
+		dockerClient, err := docker.NewClientFromEnv()
+		if err != nil {
+			return nil, err
+		}
+
+		singleInstance = &Docker{dockerClient}
+	}
+
+	return singleInstance, nil
+}
+
+func (d *Docker) CheckImagesExist(requiredImages ...string) error {
```

Comment:
Please, add a comment to any new function introduced.

Reply:
added a few comments where I find it useful. Note that we actually do not have new functions here. I was just moving code around and reuse existing things.

Reply:
yeah, but still 😅